addons/l10n_in_edi/models/account_edi_format.py

```diff
@@ -298,7 +298,7 @@ def _get_l10n_in_edi_partner_details(self, partner, set_vat=True, set_phone_and_
             partner_details.update({"POS": pos_state_id.l10n_in_tin or ""})
         if set_vat:
             partner_details.update({
-                "LglNm": partner.commercial_partner_id.name,
+                "LglNm": partner.commercial_partner_id.ref_company_ids.l10n_in_trade_name or partner.commercial_partner_id.name,
```

Comment:
Why not partner_id.commercial_partner_id.ref_company_ids.parent_id.name (or until there is no parent_id anymore)?